include/core/mu_parser_internal.h

```diff
@@ -0,0 +1,206 @@
+
+    double
+    mu_pow(double a, double b);
+
+    double
+    mu_erf(double value);
+
+    double
+    mu_erfc(double value);
+
+    double
+    mu_rand_seed(double seed);
+
+    double
+    mu_rand();
+
+    /**
+     * @brief Get the array of all function names.
+     * @return Available implemented functions
+     */
+    std::vector<std::string>
+    get_function_names();
+
+    DeclException2(ExcParseError,
+                   int,
+                   std::string,
+                   << "Parsing Error at Column " << arg1
+                   << ". The parser said: " << arg2);
+
+
+    /**
+     * @brief Base class used for thread safe instances of the parser.
+     */
+    class muParserBase
+    {
+    public:
+      virtual ~muParserBase() = default;
+    };
+
+    struct ParserData
+    {
+      ParserData() = default;
+
+      ParserData(const ParserData &) = delete;
+
+      std::vector<double> vars;
+
+      std::vector<std::unique_ptr<muParserBase>> parsers;
+    };
+
+    /**
+     * @brief Class that interfaces with muParser
+     * @tparam n_components
+     */
+    template <int n_components>
+    class ParserImplementation
+    {
+    public:
+      ParserImplementation();
+
+      virtual ~ParserImplementation() = default;
+
+      /**
+       * @brief Treat the input and initialize the muParser object
+       * @param[in] vars Variable names
+       * @param[in] expressions Expressions
+       * @param[in] constants Constants
+       * @param[in] time_dependent Boolean to define if the function is time
+       * dependent
+       */
+      void
+      initialize(const std::string                   &vars,
+                 const std::vector<std::string>      &expressions,
+                 const std::map<std::string, double> &constants,
+                 const bool                           time_dependent = false);
+
+      /**
+       * @brief Actual initialization of the muParser object
+       */
+      void
+      init_muparser() const;
+
+      /**
+       * @brief Evaluation of the function at a specific point
+       * @param[in] p Evalution point
+       * @param[in] time Current time
+       * @param[in] component Component of interest
+       * @return Value
+       */
+      double
+      do_value(const dealii::Tensor<1, n_components> &p,
+               const double                           time,
+               unsigned int                           component) const;
+
+      /**
+       * @brief Evaluation of the function at a specific point for all components
+       * @param[in] p Evaluation point
+       * @param[in] time Current time
+       * @param[out] values Output values
+       */
+      void
+      do_all_values(const dealii::Tensor<1, n_components> &p,
+                    const double                           time,
+                    dealii::ArrayView<double>             &values) const;
+
+      std::vector<std::string> expressions;
+
+    private:
+      // Thread safe object for using muParser
```

Comment:
The documentation of all the member variables of all classes should have `///` so that it appears correctly in doxygen